### ISSUE TITLE
this problems occured

### ISSUE THREAD

**Issue by superJack2222** [Junjue-Wang/LoveDA#21]
module 'ever' has no attribute 'viz'

**Comment by Junjue-Wang** [comment]
Please refer to this: https://github.com/Junjue-Wang/LoveDA/issues/5#issue-1043413685
I have already fixed this bug and you can update the proj.

**Comment by superJack2222** [comment]
but i have download the new project

**Comment by Junjue-Wang** [comment]
Please check the implementation:
https://github.com/Junjue-Wang/LoveDA/blob/master/Semantic_Segmentation/module/viz.py

**Comment by superJack2222** [comment]
i also check this ,download the code just minute ,and override the code .but also occur this problem

**Comment by superJack2222** [comment]
![image](https://user-images.githubusercontent.com/19429672/144713566-b4d5b213-6831-4221-887c-2ef4bbe1b3da.png)

**Comment by Junjue-Wang** [comment]
Please check your train.py import the wrong path？ kingdrone ***@***.*** 签名由 网易邮箱大师 定制 On 12/04/2021 22:37, superJack2222 wrote: i also check this ,download the code just minute ,and override the code .but also occur this problem — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by superJack2222** [comment]
import ever as er
import torch
import numpy as np
import os
from data.loveda import COLOR_MAP
from tqdm import tqdm
import random
from module.tta import tta, Scale


er.registry.register_all()


def evaluate_cls_fn(self, test_dataloader, config=None):
    self.model.eval()
    classes = self.model.module.config.classes if self.model.module.config.classes != 1 else 2
    metric_op = er.metric.PixelMetric(classes, logdir=self._model_dir, logger=self.logger)

    vis_dir = os.path.join(self._model_dir, 'vis-{}'.format(self.checkpoint.global_step))

    palette = np.array(list(COLOR_MAP.values())).reshape(-1).tolist()
    viz_op = er.viz.VisualizeSegmm(vis_dir, palette)

    with torch.no_grad():
        for img, gt in tqdm(test_dataloader):
            img = img.to(torch.device('cuda'))
            y_true = gt['cls']
            y_true = y_true.cpu()
            if config.get('tta', False):
                pred = tta(self.model, img, tta_config=[
                    Scale(scale_factor=0.5),
                    Scale(scale_factor=0.75),
                    Scale(scale_factor=1.0),
                    Scale(scale_factor=1.25),
                    Scale(scale_factor=1.5),
                    Scale(scale_factor=1.75),
                ])
            else:
                pred = self.model(img)
            pred = pred.argmax(dim=1).cpu()

            valid_inds = y_true != -1
            metric_op.forward(y_true[valid_inds], pred[valid_inds])

            for clsmap, imname in zip(pred, gt['fname']):
                viz_op(clsmap.cpu().numpy().astype(np.uint8), imname.replace('tif', 'png'))
    metric_op.summary_all()
    torch.cuda.empty_cache()


def register_evaluate_fn(launcher):
    launcher.override_evaluate(evaluate_cls_fn)



def seed_torch(seed=2333):
    random.seed(seed)
    os.environ['PYTHONHASHSEED'] = str(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    torch.cuda.manual_seed(seed)
    torch.cuda.manual_seed_all(seed) # if you are using multi-GPU.
    torch.backends.cudnn.benchmark = False
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.enabled = False



if __name__ == '__main__':
    seed_torch(2333)
    trainer = er.trainer.get_trainer('th_amp_ddp')()
    trainer.run(after_construct_launcher_callbacks=[register_evaluate_fn])

**Comment by superJack2222** [comment]
i copy your code

**Comment by Junjue-Wang** [comment]
you did not update the train.py …

**Comment by Junjue-Wang** [comment]
from module.viz import VisualizeSegmm

**Comment by superJack2222** [comment]
no your code is this

**Comment by superJack2222** [comment]
![image](https://user-images.githubusercontent.com/19429672/144714046-29d750fe-27fa-4af7-9bcb-b62f6d5cbb83.png)


this is your new code

**Comment by Junjue-Wang** [comment]
Sorry for this confusion. I have already revised this import bug. Thank you.
https://github.com/Junjue-Wang/LoveDA/blob/0c65e6985c7c52e503498891606f3233249f9ff1/Semantic_Segmentation/train.py

**Comment by Junjue-Wang** [comment]
Now you can re-download this for training~

**Comment by superJack2222** [comment]
ok i will try it

**Comment by Junjue-Wang** [comment]
Please let me know if it works well. Thank you!

**Comment by superJack2222** [comment]
![image](https://user-images.githubusercontent.com/19429672/144731415-b847bfce-ba31-4bd1-b495-d369056add0d.png)

**Comment by superJack2222** [comment]
how can i work in single-cpu

**Comment by Junjue-Wang** [comment]
Please refer: https://github.com/Junjue-Wang/LoveDA/blob/0c65e6985c7c52e503498891606f3233249f9ff1/Semantic_Segmentation/scripts/train_deeplabv3p.sh

**Comment by superJack2222** [comment]
![Uploading image.png…]()
it is does not effect to it

**Comment by superJack2222** [comment]
![image](https://user-images.githubusercontent.com/19429672/144734137-8450c421-d812-4752-a4cd-7841d4d8f7b2.png)

**Comment by Junjue-Wang** [comment]
It seems like an environmental deployment problem.

**Comment by superJack2222** [comment]
i run on my centos ,is this can run only cpu?

**Comment by Junjue-Wang** [comment]
This can only run on GPU. If you want to run on CPU, you should modify some codes.
